tests/testthat/test-recursive_feature_elimination.R

```diff
@@ -0,0 +1,65 @@
+#
+# Tests for recursive_feature_elimination()
+#
+
+#' Sample custom scorer that fits a model and returns
+#' an appropriate score for the feature selection problem.
+#'
+#' @param data A data.frame with features and response columns.
+#'
+#' @return The column name to be eliminated
+scorer <- function(data) {
+  model <- lm(Y ~ ., data)
+  names(which.min(model$coefficients[-1]))[[1]]
+}
+
+# This test creates a dataset that has 5 features that are are used to compute `Y`.
+# The remaining features are independent of `Y`.
+# This test should select the 5 feature columns used to compute `Y`.
+testthat::test_that("relevant features remain", {
+  # create dataframe and remove Ytrue column; keep Y.
+  set.seed(0)
+  data <- dplyr::select(tgp::friedman.1.data(), -Ytrue)
+
+  # test
+  features <- recursive_feature_elimination(scorer, data, n_features_to_select = 5)
+  testthat::expect_identical(features, c("X1", "X2", "X4", "X5", "X8", "Y"))
```

Comment:
As per discussion in python repo, recommend to change to 4 to avoid erroneous selection of X8.

```suggestion
  features <- recursive_feature_elimination(scorer, data, n_features_to_select = 4)
  testthat::expect_identical(features, c("X1", "X2", "X4", "X5", "Y"))
```